script/tool/lib/src/license_check_command.dart

```diff
@@ -0,0 +1,162 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:file/file.dart';
+import 'package:path/path.dart' as p;
+
+import 'common.dart';
+
+const Set<String> _codeFileExtensions = <String>{
+  '.c',
+  '.cc',
+  '.cpp',
+  '.dart',
+  '.h',
+  '.java',
+  '.m',
+  '.mm',
+  '.swift',
+  '.sh',
+};
+
+// Basenames without extensions of files to ignore.
+const Set<String> _ignoreBasenameList = <String>{
+  'flutter_export_environment',
+  'GeneratedPluginRegistrant',
+  'generated_plugin_registrant',
+};
+
+// File suffixes that otherwise match _codeFileExtensions to ignore.
+const Set<String> _ignoreSuffixList = <String>{
+  '.g.dart', // Generated API code.
+  '.mocks.dart', // Generated by Mockito.
+};
+
+// Full basenames of files to ignore.
+const Set<String> _ignoredFullBasenameList = <String>{
+  'resource.h', // Generated by VS.
+};
+
+// Copyright and license regexes.
+//
+// These are intentionally very simple, since almost all source in this
+// repository should be using the same license text, comment style, etc., so
+// they shouldn't need to be very flexible. Complexity can be added as-needed
+// on a case-by-case basis.
+final RegExp _copyrightRegex = RegExp(r'^(?://|#) Copyright', multiLine: true);
+// All Flutter-authored code.
+final RegExp _bsdLicenseRegex = RegExp(
+    r'^(?://|#) Use of this source code is governed by a BSD-style license',
+    multiLine: true);
```

Comment:
Can we use a static string rather than a pattern, to match our license block?

Reply:
We could, but it would mean duplicating it for bash vs. other files. What's the issue with the pattern?

(I can put the full text in the pattern, if that's the goal.)

Reply:
Seems like it would be simpler, but I see what you're saying.
FWIW this is what the flutter/flutter repo does: https://github.com/flutter/flutter/blob/master/dev/bots/analyze.dart#L226-L271
It does seem shorter than what we have here and handles bash etc.

Putting the full text in the pattern would be good regardless.

Reply:
Made it stricter as a starting point; it checks the full text of both lines, without any whitespace variation allowed.

Reply:
Since we have a bunch of variants (four different author phrasings, different years, a handful with commas after the year) I think the static string version would be too messy right now.

This PR did inspire me to continue my planned series of PRs to standardize this whole repo though (derailed by not having sign-off from repo owners to add AUTHORS to all plugins), so I'll pick that back up after this lands, and tighten this down as I go until it's at the point where we can use a string.

Reply:
I updated to use a string (using similar code to the flutter/flutter code linked above) for the license part, but not yet the copyright part. I'll switch copyright over once I've standardized.